pkg/alertmanager/alertstore/local/store_test.go

```diff
@@ -0,0 +1,133 @@
+package local
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alertspb"
+)
+
+func TestStore_ListAllUsers(t *testing.T) {
+	ctx := context.Background()
+	store, storeDir := prepareLocalStore(t)
+
+	// The storage is empty.
+	{
+		users, err := store.ListAllUsers(ctx)
+		require.NoError(t, err)
+		assert.Empty(t, users)
+	}
+
+	// The storage contains some users.
+	{
+		user1Cfg := prepareAlertmanagerConfig("user-1")
+		user2Cfg := prepareAlertmanagerConfig("user-2")
+		require.NoError(t, ioutil.WriteFile(filepath.Join(storeDir, "user-1.yaml"), []byte(user1Cfg), os.ModePerm))
+		require.NoError(t, ioutil.WriteFile(filepath.Join(storeDir, "user-2.yaml"), []byte(user2Cfg), os.ModePerm))
+
+		// The following file is expected to be skipped.
+		require.NoError(t, ioutil.WriteFile(filepath.Join(storeDir, "user-3.unsupported-extension"), []byte{}, os.ModePerm))
+
+		users, err := store.ListAllUsers(ctx)
+		require.NoError(t, err)
+		assert.ElementsMatch(t, []string{"user-1", "user-2"}, users)
+	}
+}
+
+func TestStore_GetAlertConfig(t *testing.T) {
+	ctx := context.Background()
+	store, storeDir := prepareLocalStore(t)
+
+	// The user has no config.
+	{
+		_, err := store.GetAlertConfig(ctx, "user-1")
+		assert.Equal(t, alertspb.ErrNotFound, err)
+	}
+
+	// The user has a config
+	{
+		user1Cfg := prepareAlertmanagerConfig("user-1")
+		user2Cfg := prepareAlertmanagerConfig("user-2")
+		require.NoError(t, ioutil.WriteFile(filepath.Join(storeDir, "user-1.yaml"), []byte(user1Cfg), os.ModePerm))
+		require.NoError(t, ioutil.WriteFile(filepath.Join(storeDir, "user-2.yaml"), []byte(user2Cfg), os.ModePerm))
```

Comment:
```suggestion
		user1Cfg := prepareAlertmanagerConfig("user-1")
		require.NoError(t, ioutil.WriteFile(filepath.Join(storeDir, "user-1.yaml"), []byte(user1Cfg), os.ModePerm))
```

Reply:
I don't think we need the second config for this test.

Reply:
The second config has been added to make sure that the "get" loads the correct one.

Reply:
I've added an assertion on `user-2` as well.